drivers/kscan/ec_matrix_settings.c

```diff
@@ -70,7 +85,10 @@ static void save_cb(const struct device *dev, struct zmk_kscan_ec_matrix_calibra
 
 #if IS_ENABLED(CONFIG_ZMK_KSCAN_EC_MATRIX_SETTINGS_DISCRETE)
     for (size_t i = 0; i < len; i++) {
-        snprintf(setting_name, MAX_SETTING_LEN, "zmk/ec/calibration/%s/%d", dev->name, i);
+        int wl = snprintf(setting_name, MAX_SETTING_LEN, "zmk/ec/calibration/%s/%d", dev->name, i);
+        if (wl >= MAX_SETTING_LEN) {
```

Comment:
The check here doesn't match the earlier check, that subtracts the `DISCRETE_SETTING_SUFFIX_LEN`. That seems like an error. Am I missing something?